llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -79,6 +79,8 @@ bool VPRecipeBase::mayWriteToMemory() const {
     return !cast<VPWidenCallRecipe>(this)
                 ->getCalledScalarFunction()
                 ->onlyReadsMemory();
+  case VPWidenIntrinsicSC:
+    return false;
```

Comment:
This might be incorrect for memaccesses intrinsics. Need somehow to restrict the class for using with such intrinsics

Reply:
Yep, updated to store read/write/sideeffcts in recipe.